Декоративные элементы всех страниц

SerialEditor opened the pull request:
---
:mortar_board: [Декоративные элементы всех страниц](https://up.htmlacademy.ru/htmlcss-individual/1/user/1745893/tasks/10)

:boom: https://htmlacademy-htmlcss.github.io/1745893-sedona-1/8/

SerialEditor committed:
Добавляет декоративные стили

Добавляет декоративные стили элементам шапки и подвала сайта, элементам главной страницы

SerialEditor committed:
Оформляет декоративные элементы

Оформляет декоративные элементы и интерактивные состояния интерфейса страниц сайта

SerialEditor committed:
Добавляет интерактивные стили элементам

SerialEditor committed:
Дополняет декоративные стили элементов интерфейса

SerialEditor commented on images/icons/block-layout.svg:
Проморгал удаление viewbox у векторных изображений при оптимизации SVGOMG - экспортировал заново; в нескольких редактировал код (добавил удалённый viewbox)

SerialEditor commented on styles/styles.css:
```diff
@@ -111,22 +153,81 @@ body {
   text-decoration: none;
 }
 
+.site-navigation-link[href]:hover {
```
В макете нет описания пользовательского взаимодействия со ссылками в навигации (.site-navigation-link, user-navigation-link) - задал стили, старался попасть в тон макету

andreysgra commented on styles/styles.css:
```diff
@@ -52,6 +52,42 @@ body {
+  background-color: #615048;
+}
+
+.button:focus {
```
Здесь, и для других интерактивных элементов, состояние фокуса лучше описывать через псевдо класс `:focus-visible`. Отличие этого псевдо класса от `:focus` в том, что браузер сам решает, нужно ли визуальное изменение на этом элементе в состоянии фокуса. Если для навигации используется клавиатура, то состояние фокуса отображается, если же мышь, то нет, так как пользователям использующим мышь неважно находится ли данный элемент в фокусе.

Визуальное различие при использовании мыши можно заметить таким образом: нажмите кнопку мыши на интерактивном элементе, где стили фокуса описаны через псевдо класс `:focus` и не отпуская кнопку уведите курсор за пределы области этого элемента, а затем отпустите кнопку. На этом элементе останется визуальное состояние фокуса. Если же описать все эти свойства в `:focus-visible`, то такого эффекта не будет (сам эффект на мой взгляд нежелателен).

andreysgra commented on styles/styles.css:
```diff
@@ -111,22 +153,81 @@ body {
+  background: #ffffff url("../images/icons/heart.svg") no-repeat center;
+}
+
+.user-navigation-link-search:focus, .user-navigation-link-favorite:focus {
+  background-color: rgba(131, 179, 211, 0.2);
+  outline: none;
+}
+
+.user-navigation-link-search:active, .user-navigation-link-favorite:active {
+  background-color: #ffffff;
+  opacity: 0.3;
+}
```
[PROJ-01](https://up.htmlacademy.ru/htmlcss-individual/1/criteries#PROJ-01): в этом [пункте](https://codeguide.academy/html-css.html#css-multiselectors) показан пример переноса селекторов.

andreysgra commented on styles/styles.css:
```diff
@@ -434,18 +532,68 @@ body {
+  margin: auto;
+}
+
+.breadcrumbs-link-homepage:hover path, .breadcrumbs-link-homepage:focus path {
```
[PROJ-01](https://up.htmlacademy.ru/htmlcss-individual/1/criteries#PROJ-01)

andreysgra commented on styles/styles.css:
```diff
@@ -617,18 +800,34 @@ body {
   background-color: #82b3d3;
 }
 
+.add-favorites:hover, .add-favorites:focus {
```
[PROJ-01](https://up.htmlacademy.ru/htmlcss-individual/1/criteries#PROJ-01)

andreysgra commented on styles/styles.css:
```diff
@@ -617,18 +800,34 @@ body {
   background-color: #7db54f;
 }
 
+.added-favorites:hover, .added-favorites:focus {
```
[PROJ-01](https://up.htmlacademy.ru/htmlcss-individual/1/criteries#PROJ-01)

andreysgra commented on styles/styles.css:
```diff
@@ -674,6 +909,10 @@ body {
   background-color: #ffffff;
 }
 
+.pagination-show-more:hover, .pagination-show-more:focus {
```
[PROJ-01](https://up.htmlacademy.ru/htmlcss-individual/1/criteries#PROJ-01)

andreysgra commented on styles/styles.css:
```diff
@@ -692,11 +931,30 @@ body {
+  margin: auto;
+}
+
+.footer-social-link:hover path, .footer-social-link:focus path {
```
[PROJ-01](https://up.htmlacademy.ru/htmlcss-individual/1/criteries#PROJ-01)

andreysgra commented on styles/styles.css:
```diff
@@ -707,7 +965,32 @@ body {
+  outline: none;
+}
+
+.htmlacademy-logo:hover path, .htmlacademy-logo:focus path {
```
[PROJ-01](https://up.htmlacademy.ru/htmlcss-individual/1/criteries#PROJ-01)

SerialEditor committed:
Исправляет декоративные стили элементов интерфейса

Меняет селектор по псевдоклассу ':focus' на ':focus-visible'; приводит запись перечисляемых селекторов к соответствию кодгайду
